src/coreclr/src/inc/corprof.idl

```diff
@@ -4073,28 +4114,50 @@ interface ICorProfilerInfo11 : ICorProfilerInfo10
 ]
 interface ICorProfilerInfo12 : ICorProfilerInfo11
 {
+    HRESULT EventPipeStartSession(
+        [in]  UINT32                            cProviderConfigs,
+        [in, size_is(cProviderConfigs)]
+              COR_PRF_EVENTPIPE_PROVIDER_CONFIG pProviderConfigs[],
+        [in]  BOOL                              requestRundown,
+        [in]  BOOL                              requestCPUSamples,
+        [out] EVENTPIPE_SESSION*                pSession);
+
+    HRESULT EventPipeAddProviderToSession(
```

Comment:
We don't expose a mechanism over the pipe, but internally we have the functionality already. When we do rundown we add the rundown provider to the session.

Just like EventPipeProviderCreated above, this was added for testing (to be able to subscribe programatically to all events). But, I can imagine scenarios where it might be easier to add providers rather than starting a new session.

Reply:
Yeah I think being able to add (and for that matter change/remove) providers is goodness, I was just surprised to see it in the profiler first : )
I'm happy to keep it as long we can deliver the notification outside of the EventPipe lock.